catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -715,11 +716,25 @@ public void setAttribute(String name, Serializable value) {
 
     @Override
     public void setAttribute(Attribute attribute) {
-
         if (attribute == null) {
             return;
         }
 
+        AttributeDescriptor descriptor = this.type.getAttributeDescriptor(attribute.getName());
+        if (descriptor != null) {
+            Class binding = descriptor.getType().getBinding();
+            if (binding != null) {
+                AttributeType.AttributeFormat format = descriptor.getType().getAttributeFormat();
+                if (attribute.getValue() != null && !attribute.getValue().getClass().equals(binding)) {
```

Comment:
Has this been fixed? Or proven out to not have a difference? I'm a fan of using the more permissive `isAssignableFrom`